e2etests/node_scan_rhcos_test.go

```diff
@@ -0,0 +1,166 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// vulnLibksba is an example of a fixable vulnerability
+var vulnLibksba = &v1.Vulnerability{
+	Name:        "RHSA-2022:7089",
+	Description: "KSBA (pronounced Kasbah) is a library to make X.509 certificates as well as the CMS easily accessible by other applications.  Both specifications are building blocks of S/MIME and TLS.\n\nSecurity Fix(es):\n\n* libksba: integer overflow may lead to remote code execution (CVE-2022-3515)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.",
+	Link:        "https://access.redhat.com/errata/RHSA-2022:7089",
+	MetadataV2: &v1.Metadata{
+		PublishedDateTime:    "2022-10-24T00:00Z",
+		LastModifiedDateTime: "",
+		CvssV2:               nil,
+		CvssV3: &v1.CVSSMetadata{
+			Score:               8.1,
+			Vector:              "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:H/I:H/A:H",
+			ExploitabilityScore: 2.2,
+			ImpactScore:         5.9,
+		},
+	},
+	FixedBy:  "0:1.3.5-8.el8_6",
+	Severity: "Important",
+}
+
+// vulnTar is an example of a non-fixable vulnerability
+var vulnTar = &v1.Vulnerability{
+	Name:        "CVE-2005-2541",
+	Description: "DOCUMENTATION: The MITRE CVE dictionary describes this issue as: Tar 1.15.1 does not properly warn the user when extracting setuid or setgid files, which may allow local users or remote attackers to gain privileges. \n            STATEMENT: This CVE was assigned to what is the documented and expected behaviour of tar.  There are currently no plans to change tar behaviour to strip setuid and setgid bits when extracting archives.",
+	Link:        "https://access.redhat.com/security/cve/CVE-2005-2541",
+	MetadataV2: &v1.Metadata{
+		CvssV2: nil,
+		CvssV3: &v1.CVSSMetadata{
+			Score:               7,
+			Vector:              "CVSS:3.1/AV:L/AC:H/PR:N/UI:R/S:U/C:H/I:H/A:H",
+			ExploitabilityScore: 1,
+			ImpactScore:         5.9,
+		},
+	},
+	FixedBy:  "",
+	Severity: "Moderate",
+}
+
+func TestGRPCGetRHCOSNodeVulnerabilities(t *testing.T) {
+	conn := connectToScanner(t)
+	client := v1.NewNodeScanServiceClient(conn)
+	if !features.RHCOSNodeScanning.Enabled() {
+		t.Skipf("Feature RHCOS Node Scanning is disabled")
+	}
+
+	cases := []struct {
+		request          *v1.GetNodeVulnerabilitiesRequest
+		responseContains *v1.GetNodeVulnerabilitiesResponse
+	}{
+		{
+			request: &v1.GetNodeVulnerabilitiesRequest{
+				OsImage:          "Red Hat Enterprise Linux CoreOS 45.82.202008101249-0 (Ootpa)",
+				KernelVersion:    "0.0.1",
+				KubeletVersion:   "0.0.1",
+				KubeproxyVersion: "0.0.1",
+				Runtime: &v1.GetNodeVulnerabilitiesRequest_ContainerRuntime{
+					Name:    "docker",
+					Version: "0.0.1",
+				},
+				NodeInventory: &v1.Components{
+					Namespace: "Namespace-Name",
+					RhelComponents: []*v1.RHELComponent{
```

Comment:
Having an additional component without vulnerabilities in this test would also be very nice.